src/crashlog.cpp

```diff
@@ -55,8 +55,8 @@ static void SurveyRecentNews(nlohmann::json &json)
 {
 	json = nlohmann::json::array();
 
-	int i = 0;
-	for (NewsItem *news = _latest_news; i < 32 && news != nullptr; news = news->prev, i++) {
+	auto [first, last] = GetRecentNews(32);
+	for (auto news = first; news != last; ++news) {
```

Comment:
Is there a reason to have this function? It's only used once in here, and with this it has to iterate through the news items twice. After all `std::distance` is not a constant operation of `std::list` compared to `std::vector`.

May something inspired by the following would work?
```suggestion
	for (auto news : std::ranges::take_view(_news, 32)) {
```

Reply:
std::ranges doesn't work with clang 15 :(